coderd/httpmw/apikey.go

```diff
@@ -142,6 +142,56 @@ func ExtractAPIKeyMW(cfg ExtractAPIKeyConfig) func(http.Handler) http.Handler {
 	}
 }
 
+func APIKeyFromRequest(ctx context.Context, db database.Store, sessionTokenFunc func(r *http.Request) string, r *http.Request) (*database.APIKey, codersdk.Response, bool) {
```

Comment:
Instead of this could you have just added a field to the opts struct above "NoRefreshToken" or something?

Reply:
I could yea, but I didn't want to keep using that middleware for just the key extraction. It could evolve more features overtime and the use case in this situation doesn't ever change/evovle.